test/built-ins/Promise/prototype/finally/subclass-species-constructor-reject-count.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2018. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: finally on rejected Promise calls the SpeciesConstructor
+esid: sec-promise.prototype.finally
+features: [Promise.prototype.finally]
+---*/
+
+class FooPromise extends Promise {
+  static get [Symbol.species]() { return Promise; }
+}
+
+var p = Promise.reject().finally(() => FooPromise.reject());
+
+assert.sameValue(p instanceof Promise, true);
+assert.sameValue(p instanceof FooPromise, true);
```

Comment:
This should be false because `.then()` in Step 7 of `Promise.prototype.finally` is called on the receiver of `finally` which is `Promise`.